src/scripts/modal.js

```diff
@@ -0,0 +1,26 @@
+function modal() {
+  const heroContent = document.querySelector(".hero__content");
+  const modalContent = document.querySelector("#modal");
+  const checkOfferBtn = document.querySelector(".hero--cta");
+  const confrimModalBtn = modalContent.querySelector(".modal-btn--confirm");
+  const closeModalBtn = modalContent.querySelector(".modal-btn--close");
+  const closeModalIcon = modalContent.querySelector(".modal-icon--close");
+
+  function close() {
+    modalContent.classList.remove("modal--show");
+    heroContent.classList.remove("blur");
+  }
+  checkOfferBtn.addEventListener("click", () => {
+    modalContent.classList.add("modal--show");
+    heroContent.classList.add("blur");
+  });
+
+  [closeModalBtn, closeModalIcon].forEach((closeModal) =>
+    closeModal.addEventListener("click", close)
+  );
+
+  document.addEventListener("mousedown", (e) => {
+    modalContent.contains(e.target) ? "" : close();
+  });
```

Comment:
it should be on `click`, not `mousedown`

Reply:
@FrostEvil we still miss the clean up... When modal is closed this listener should be removed

Reply:
let's take a `console.log("hello")` as callback example.
When modal is opened we set a listener, so every click would produce a message in the console.
Let's say we close the modal and open it again - another listener is being registered. Now on **each click** we fire the callback **twice** (two messages will appear). This is not the expected behaviour, so we have to remove the listener each time modal is closed 😉

Reply:
changed